examples/code-dom-injection/content.js

```diff
@@ -0,0 +1,37 @@
+// Enable HMR if available
+if (module.hot) {
```

Comment:
to have it working I had to add a wrapper script that imports the module, but I'm not sure about the implications of using it with `webpack-target-webextension` or the react reloader for example so I appreciate any guidance

```ts
// minimum-content-script-file.js - Wrapper for user-defined content_scripts

function loadContentScript() {
  // Import content.js dynamically so HMR updates are tracked
  return import('./content.js')
}

// Content script initial load
let cleanup = loadContentScript()

// Enable HMR for the content script
if (module.hot) {
  module.hot.accept('./content.js', () => {
    // Remove previous content
    cleanup && cleanup()

    // Re-import the updated content script and run it again
    cleanup = loadContentScript()
  })

  // Ensure cleanup on dispose
  module.hot.dispose(() => {
    cleanup && cleanup()
  })
}
```